### PR TITLE
Add signal.h

### PR DESCRIPTION
**Pull request by dlrobertson**
Create stubs for signal.h

**NB:** Duplicate definitions for `SIGINT` etc will exist in the header until https://github.com/eqrion/cbindgen/pull/148 is merged.

### CODE REVIEW

File: src/signal/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    pub struct sigaction {
+        pub sa_handler: extern "C" fn(usize),
+        pub sa_mask: sigset_t,
+        pub sa_flags: usize,
+    }
+}
+
+// RedoxOS Signals
+
+#[cfg(target_os = "redox")]
```
Comment (jackpot51) [inline src/signal/src/lib.rs]: Look at fcntl for a better way to do this

Comment (jackpot51) [reply src/signal/src/lib.rs]: A separate mod for Redox and Linux

Comment (dlrobertson) [reply src/signal/src/lib.rs]: AFAIK the constants don't get exported into the header with this though. I don't see the `O_RDONLY` constant in `fcntl.h`.

Comment (dlrobertson) [reply src/signal/src/lib.rs]: I agree that using a sys crate is **much** more readable though. Is there a way to implement these like the `fcntl` crate and export the constants. Or is the fact that the constants are not exported a bug in `cbindgen`?

Comment (jackpot51) [reply src/signal/src/lib.rs]: I will look into it

**Pull request merged**